Lib/pprint.py

```diff
@@ -141,8 +141,13 @@ def __init__(self, indent=1, width=80, depth=None, stream=None, *,
         self._width = width
         if stream is not None:
             self._stream = stream
-        else:
+        elif _sys.stdout is not None:
             self._stream = _sys.stdout
+        else:
+            class _NullStdout:
+                def write(self, s):
+                    return 0
+            self._stream = _NullStdout()
```

Comment:
Why bother with all of the formatting work if nothing will ever be written to a real output stream? Wouldn't it be simpler to change the public `pprint()` method to check for `None`? For example:

```python
def pprint(self, object):
    if self._stream is not None:
        self._format(object, self._stream, 0, 0, {}, 0)
        self._stream.write("\n")
```

The private methods that use the stream should never be called directly, so only the `pprint()` method should have to change.

This matches how builtin `print()` works:

```C
    if (file == NULL || file == Py_None) {
        file = _PySys_GetObjectId(&PyId_stdout);
        if (file == NULL) {
            PyErr_SetString(PyExc_RuntimeError, "lost sys.stdout");
            return NULL;
        }

        /* sys.stdout may be None when FILE* stdout isn't connected */
        if (file == Py_None)
            Py_RETURN_NONE;
    }
```

Reply:
Yes, good point!